### PR TITLE
Change location

### PR DESCRIPTION
**Pull request by Almightywesley**
Yoo guys 
 i made the change memory location please revieuw it.
you can add a memory by long pressing and by searching in the searchbar and you can save the memory

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -22,4 +22,5 @@
     <string name="action_ok">Got it!</string>
     <string name="error_could_not_set_current_location">Could not set memory location to current.</string>
     <string name="action_apply_theme">Apply Theme</string>
+    <string name="action_change_memory_location">Change memory location</string>
```
Comment (MrTheGood) [inline app/src/main/res/values/strings.xml]: I think this should be changed from `Change memory location` to `Change location`

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java [hunk 2]
```diff
@@ -43,6 +55,8 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_memory_edit);
 
+
+
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java:58]: These are some beautiful lines. Amazing.

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java [hunk 3]
```diff
@@ -1,36 +1,48 @@
...
 import nl.hogeschoolrotterdam.projectb.util.LocationManager;
 import nl.hogeschoolrotterdam.projectb.util.SimpleTextWatcher;
 
-import java.util.ArrayList;
-import java.util.Calendar;
-import java.util.Date;
+import java.util.*;
 
 public class MemoryEditActivity extends AppCompatActivity {
+    private static final String TAG = "AutoCompleteAdapter";
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java]: This tag isn't used anywhere.

Comment (MrTheGood) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java]: So it should be removed.

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+            public void onClick(View v) {
+                Intent returnIntent = new Intent();
+                returnIntent.putExtra("result",latLng);
+                setResult(Activity.RESULT_OK,returnIntent);
+                finish();
+
+            }
+        });
+        mMapView = findViewById(R.id.map);
+        if (mMapView != null) {
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: mMapView is never null. This `if` statement is useless.

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        List<Address> list = new ArrayList<>();
+        try {
+            list = geocoder.getFromLocationName(searchingString, 1);
+
+
+        } catch (IOException e) {
+        }
+        if (list.size() > 0) {
+            Address address = list.get(0);
+            Log.d(TAG, "Locate adress:" + address.toString());
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: This log could be removed. We don't need to spam logcat.

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+            //Toast.makeText(this, address.toString(), Toast.LENGTH_SHORT).show();
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: This comment could be removed.

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+            CameraPosition search = CameraPosition.builder().target(new LatLng(address.getLatitude(),
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: Shouldn't this `CamaeraPosition` thing be moved to the `setMarker` method?

Comment (MrTheGood) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: The camera position should always be moved to the current marker, so if you move these 2 lines to `setMarker`, it'll always happen.

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+        mGoogleMap.setOnMapLongClickListener(new GoogleMap.OnMapLongClickListener(){
+
+            @Override
+            public void onMapLongClick(LatLng Longl) {
+                latLng = Longl;
+                setMarker(latLng);
+
+            }
+        });
+        mSearchText = (EditText) findViewById(R.id.input_search);
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: There is a useless cast here. `(EditText)` could be removed.

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+/**
+ * Created by maartendegoede on 20/03/2019.
+ * Copyright © 2019 Anass El Mahdaoui, Hicham El Marzgioui, Michaël van Asperen, Wesley de Man, Maarten de Goede all rights reserved.
+ */
+public class LocationEditActivity extends AppCompatActivity implements OnMapReadyCallback {
+
+    GoogleMap mGoogleMap;
+    MapView mMapView;
+    private EditText mSearchText;
+    private static final String TAG = "Location";
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: If you remove the log below (see a later comment), then this `TAG` constant could be removed.

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+import nl.hogeschoolrotterdam.projectb.data.Database;
+import nl.hogeschoolrotterdam.projectb.data.Memory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by maartendegoede on 20/03/2019.
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: I did not create this class. You may change this to your name :)
Also: It's 24/04/2019

File: app/src/main/res/layout/activity_location_edit.xml [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+            android:hint="@string/str_enter_adress"
+            android:textSize="15sp"
+            android:singleLine="true"
+            android:imeOptions="actionSearch"
+            />
+
+
+    </RelativeLayout>
+
+    <Button
```
Comment (MrTheGood) [inline app/src/main/res/layout/activity_location_edit.xml:49]: I think this button should have a margin bottom, to move it up a tiny bit. Let's say a margin of `32dp` or `16dp`

**Review by MrTheGood — CHANGES_REQUESTED** [review]
Nice work! 
I have added some change requests, though.

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java (hunk 4 above)
Comment (Almightywesley) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: How do you know?

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java (hunk 5 above)
Comment (Almightywesley) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: this is for testing purposses

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (Almightywesley) [reply app/src/main/res/values/strings.xml]: done

**Review by AnassM06 — APPROVED** [review]
looks good

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java (hunk 3 above)
Comment (Almightywesley) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java]: done

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java (hunk 4 above)
Comment (Almightywesley) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: done

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java (hunk 5 above)
Comment (Almightywesley) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: done

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java (hunk 6 above)
Comment (Almightywesley) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: done

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java (hunk 7 above)
Comment (Almightywesley) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: done

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java (hunk 8 above)
Comment (Almightywesley) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: done

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java (hunk 9 above)
Comment (Almightywesley) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: done

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java (hunk 10 above)
Comment (Almightywesley) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: done

File: app/src/main/res/layout/activity_location_edit.xml (hunk 11 above)
Comment (Almightywesley) [reply app/src/main/res/layout/activity_location_edit.xml:49]: done

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+        mGoogleMap.clear();
+        MarkerOptions options = new MarkerOptions().position(latLng);
+        mGoogleMap.addMarker(options);
+
+    }
+
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java:97]: Above `super.onCreate(savedInstanceState);`, please add `setTheme(WhibApp.getInstance().getThemeId());` to correctly set the theme of the activity.

**Review by MrTheGood — CHANGES_REQUESTED** [review]

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java (hunk 5 above)
Comment (MrTheGood) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: Yes but we don't need to log this anymore.

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java (hunk 4 above)
Comment (MrTheGood) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java]: Because `mMapView` is set in the line above it.

**Review by MrTheGood — APPROVED** [review]

**Pull request merged**